src/wblib/figures/internal/olr.py

```diff
@@ -0,0 +1,115 @@
+import intake
+import numpy as np
+import xarray as xr
+import pandas as pd
+
+import cartopy.crs as ccrs
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.figure import Figure
+import matplotlib.colors as mcolors
+import seaborn as sns
+import easygems.healpix as egh
+
+from wblib.figures.hifs import get_latest_forecast_issue_time
+
+
+CATALOG_URL = "https://tcodata.mpimet.mpg.de/internal.yaml"
+CATALOG_OLR_CODE = "ttr"
+
+OLR_MAX = 250
+OLR_MIN = 100
+
+FIGURE_SIZE = (15, 8)
+FIGURE_BOUNDARIES = (-70, 10, -10, 30)
+
+
+def toa_outgoing_longwave(
+    briefing_time: pd.Timestamp, lead_hours: str, *args, **kwargs
+) -> Figure:
+    lead_delta = pd.Timedelta(hours=int(lead_hours[:-1]))
+    issued_time = get_latest_forecast_issue_time(briefing_time)
+    valid_time = briefing_time + lead_delta
+    accumulated_ttr = _get_ttr_forecast_datarray(issued_time)
+    olr = _convert_accumulated_ttr_to_olr(accumulated_ttr, issued_time)
+    olr = olr.sel(time=valid_time.tz_localize(None))
+
+    # plot
+    sns.set_context("talk")
+    fig, ax = plt.subplots(
+        figsize=FIGURE_SIZE, subplot_kw={"projection": ccrs.PlateCarree()}
+    )
+    _draw_olr(olr, fig, ax)
+    _format_axes(briefing_time, issued_time, lead_delta, ax)
+    matplotlib.rc_file_defaults()
+    return fig
+
+
+def _get_ttr_forecast_datarray(issued_time: pd.Timestamp) -> xr.DataArray:
+    catalog = intake.open_catalog(CATALOG_URL)
```

Comment:
And the catalog is something that I think we should also take outside of the functions, so that this is passed to them instead of initialized within them. Yet, also a topic of a different pull request.

Reply:
I agree.